.github/workflows/build_steps.yml

```diff
@@ -146,6 +146,41 @@ jobs:
         env:
           ARCTICDB_memory_loglevel: INFO
 
+      - name: Select Python (Linux)
```

Comment:
Why not call the actual `python_tests` job instead of hand-rolling another?

Reply:
I am having some problems with running the python_tests job on the ec2 machines.
Also with the way it is design, it would run slower on the ec2 machines, than this solution.

Reply:
Could you share what problems you saw?

> Also with the way it is design, it would run slower on the ec2 machines, than this solution.

I presume this is because each job has to wait in turn for the only runner instance. To resolve this, you can use Docker to start multiple Github Action runners on one machine.

Also, another thing that the existing `python_tests` job takes care of is to have a realistic environment that's fully isolated from the build directories and each other. For example, it will install the actual packaged wheel in a fresh venv.

Reply:
I think that I was having some problems with building the wheels, I will try to reproduce later today.

Yeas, each job would have to wait, but also just building and installing the wheel takes around as much time as just running the tests like I have done here.
I am still in favor of doing it this way, as would be simpler and faster.

Reply:
I think Georgi's intent with this PR is to introduce a fast "canary" build. It builds on one Python and runs the tests on it as fast as it can. Once it passes, developers can be quite confident their changes are fine, and move on to something else, without waiting for all the other things in the build. I think it makes sense.

Reply:
> I think Georgi's intent with this PR is to introduce a fast "canary" build

That is pretty clear from the code; however, this design only speeds up these discoveries:

1. The Python module doesn't load
2. Some test(s) fail for the chosen Python version

But it will slow down/delay the discovery of these:

3. Some test(s) fail in other Python version(s) jobs
4. Some test(s) become flaky


Based on my experience, most people will run the relevant tests locally during development, and it's rare that code changes in one area break a test in a completely unexpected area, so speeding up 2 provides limited benefits.

However, under this "single-job on the fast machine" design, if we get a flaky test, it's no longer possible to retry the tests without restarting the whole build (including the Windows parts) because of how the fast build machine is started/stopped. With a bit of refactoring, it should be possible to run one branch of the existing test jobs on it and retain the ability to retry just one kind of test.

Reply:
The test step has been added below. and it delaying the execution of the wheel builds.
This is what we had agreed on when we last discussed it. because this way it would be the fastest (wouldn't require any more recompiles) and it would stop the build as fast as possible, if there are any failing tests.
And it is not actually slowing down the total build as that is bound by the windows builds.
 
Do you want me to add it as a separate, non blocking job, similar to the C++ tests?